azurerm/internal/services/maintenance/maintenance_configuration_resource.go

```diff
@@ -62,26 +63,75 @@ func resourceArmMaintenanceConfiguration() *pluginsdk.Resource {
 			"scope": {
 				Type:     pluginsdk.TypeString,
 				Optional: true,
-				Default:  string(maintenance.ScopeAll),
+				Default:  "All",
 				ValidateFunc: validation.StringInSlice([]string{
-					string(maintenance.ScopeAll),
+					"All", // All is still accepted by the API
+					string(maintenance.ScopeExtension),
 					string(maintenance.ScopeHost),
-					string(maintenance.ScopeInResource),
-					string(maintenance.ScopeResource),
+					string(maintenance.ScopeInGuestPatch),
+					string(maintenance.ScopeOSImage),
+					string(maintenance.ScopeSQLDB),
+					string(maintenance.ScopeSQLManagedInstance),
 				}, false),
 			},
 
-			// There's a bug in the Azure API where the the key of tags is returned in lower-case
-			// BUG: https://github.com/Azure/azure-rest-api-specs/issues/9075
-			// use custom tags defition here to prevent inputting upper case key
-			"tags": {
-				Type:         pluginsdk.TypeMap,
-				Optional:     true,
-				ValidateFunc: validate.TagsWithLowerCaseKey,
+			"visibility": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Default:  string(maintenance.VisibilityCustom),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(maintenance.VisibilityCustom),
+					// Creating public configurations doesn't appear to be supported, API returns `Public Maintenance Configuration must set correct properties`
+					// string(maintenance.VisibilityPublic),
+				}, false),
+			},
+
+			"window": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"start_date_time": {
+							Type:     pluginsdk.TypeString,
+							Required: true,
+						},
+						"expiration_date_time": {
```

Comment:
for consistancy can we change tis to
```suggestion
						"end_date_time": {
```

Reply:
So my understanding is this is the expiration date for the maintenance window overall, not the end time of each iteration, if that makes sense?

> Effective expiration date of the maintenance window in YYYY-MM-DD hh:mm format. The window will be created in the time zone provided and adjusted to daylight savings according to that time zone. Expiration date must be set to a future date. If not provided, it will be set to the maximum datetime 9999-12-31 23:59:59.